### DISCUSSION TITLE
`bg-gradient-to-r` fallback behaviour different in v3 from v2

### DISCUSSION THREAD

**Discussion by mishig25** [tailwindlabs/tailwindcss#9221] [Help] [answered]
let's say I have this button with tailwind classes. As you can see, `hover:to-` is not defined, only `hover:from-gray-600` is defined
```html
<button class="bg-gradient-to-r from-gray-300 to-gray-900 hover:from-gray-600">my button</button>
```

tailwind v2 fallback to `to-gray-900` class on hover (see tailwindplay [v2 here](https://play.tailwindcss.com/mEE2PBMVE7)).

but, tailwind v3 fallback to `to-white` class on hover (see tailwindplay [v3 here](https://play.tailwindcss.com/P2WnIHxMFF)). And to get the same behaviour as v2, I need to explictly add `hover:to-gray-900` to the classlist above.

Is there any config I can supply to `tailwind.config.js` to have the v2 fallback behaviour in v3? Moreover, this difference was not described in the [upgrade guide](https://tailwindcss.com/docs/upgrade-guide). 

Thanks in advance! Really love using Tailwind with Svelte

**ps: I've used `hover` variant in this example. But the same problem is present with any other variants like `dark`, `md`, etc.

**Comment by wongjn** [comment, marked as answer]
It seems like it is because of this [feature with gradient stops](https://tailwindcss.com/docs/gradient-color-stops#fading-to-transparent) – your v3 example does not go to `white`, but to `gray-900/0` (`gray.900` fully transparent). You could reverse the gradient direction and the color stops to emulate v2, [see this example Tailwind Play](https://play.tailwindcss.com/HdwUK9ObS5). Essentially, because of the assumed `to-transparent` feature in v3, you should have the static portion as the `from-*` color stop and change the `to-*` with the variant.

**Reply by mishig25** [reply]
I see, thanks so much for explaining the issue & providing the suggestion!
If I understand it correctly, there is no config/flag I can just "turn on" to disable [fading-to-transparent](https://tailwindcss.com/docs/gradient-color-stops#fading-to-transparent) behaviour right?

**Reply by wongjn** [reply]
There does not seem to be a config or flag that I have seen.

**Reply by mishig25** [reply]
I see. Thanks so much for the help!
